### PR TITLE
Proposal: Enhance `EcternalEtcd` Configuration to Support Retrieving etcd Client Credentials from a Kubernetes Secret

### PR DESCRIPTION
**Pull request by jabellard**
**What type of PR is this?**

/kind design

**What this PR does / why we need it**:

Details are provided in the proposal.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by jabellard** [comment]
@zhzhuang-zju , thanks for the constructive feedback. I pushed a change to address comments.

File: docs/proposals/external-etcd-credentials-ref/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+### Non-Goals
+
+- Modifying the internal etcd configuration.
+- Changing the existing inline configuration approach.
+
+
+## Proposal
+
+Introduce a new field, `SecretRef`, within the `ExternalEtcd` configuration to reference a Kubernetes secret that contains the necessary etcd connection credentials. 
+The existing inline configuration fields will be preserved to maintain backward compatibility. However, given that they are currently unused, it would make sense to mark them as deprecated.
```
[the remark this answers is not in the document]
Comment (RainbowMango) [reply docs/proposals/external-etcd-credentials-ref/README.md:47]: Given those fields are unused and might bring security concerns, I think we should mark them deprecated.

File: docs/proposals/external-etcd-credentials-ref/README.md
Comment (RainbowMango) [inline docs/proposals/external-etcd-credentials-ref/README.md:1]: @jabellard Do you think we should group operator-related proposals into a separated directory? Like `docs/proposals/operator`?

**Review by RainbowMango — COMMENTED** [review]
Generally looks good to me.
Thanks @jabellard 
This feature is more like supporting external ETCD in a security gesture!

File: docs/proposals/external-etcd-credentials-ref/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+The current implementation of the `ExternalEtcd` configuration requires the etcd client credentials, including the certificate private key, to be provided inline within the configuration. 
+This approach poses security risks and can be cumbersome to manage, especially in CI/CD scenarios driven by GitOps. By enabling the ability of loading those credentials from a secret, we can ensure that sensitive data is stored securely and managed more efficiently. 
+Additionally, this enhancement is very well-aligned with Kubernetes best practices for managing sensitive information and provides a more flexible and secure way to configure external etcd connections.
+
+
+### Goals
+
+- Enable the use of Kubernetes secrets for storing and retrieving external etcd connection credentials.
+- Improve the security of sensitive data by leveraging Kubernetes' native secret management capabilities.
+- Simplify the management of external etcd configurations.
```
[the remark this answers is not in the document]
Comment (RainbowMango) [reply docs/proposals/external-etcd-credentials-ref/README.md:35]: +1

File: docs/proposals/external-etcd-credentials-ref/README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+kind: Secret
+metadata:
+  name: tenant1-etcd-credentials
+  namespace: tenant1
+type: Opaque
+data:
+  ca.crt: <base64-encoded-data>
+  tls.crt: <base64-encoded-data>
+  tls.key: <base64-encoded-data>
+```
```
[the remark this answers is not in the document]
Comment (RainbowMango) [reply docs/proposals/external-etcd-credentials-ref/README.md]: The data of the secret will be shared with `karmada-apiserver`, `karmada-aggregate-apiserver`, and `karmada-search`, which requires access to ETCD.

File: docs/proposals/external-etcd-credentials-ref/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (zhzhuang-zju) [reply docs/proposals/external-etcd-credentials-ref/README.md:47]: > Given those fields are unused and might bring security concerns, I think we should mark them deprecated

agree, also reduces maintenance costs

**Comment by jabellard** [comment]
@RainbowMango , thanks for your constructive feedback. I just pushed some changes to address your comments so I think this should be ready to go.

**Comment by zhzhuang-zju** [comment]
> The hosted runner: GitHub Actions 607 lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.

the `lint` failed for some unknown reason
/retest

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**